spyder/workers/updates.py

```diff
@@ -95,5 +96,7 @@ def start(self):
         except Exception:
             error_msg = _('Unable to check for updates.')
 
-        self.error = error_msg
-        self.sig_ready.emit()
+        # Don't show dialog when starting up spyder and an error occur
+        if not self.startup or error_msg is None:
```

Comment:
Yes, that's the idea, the expression will evaluate `True` when startup is `False` (regardless if `error_message` is `None` or not)